pkg/test/resourcefixture/contexts/register.go

```diff
@@ -447,20 +447,30 @@ func directCreate(ctx context.Context, u *unstructured.Unstructured, c client.Cl
 }
 
 func directDelete(ctx context.Context, u *unstructured.Unstructured, c client.Client) error {
+	maxRetries := 3
+	baseDelay := 100 * time.Millisecond // starting delay
+	var err error
+
 	a, err := getAdapter(ctx, u, c)
 	if err != nil {
 		return err
 	}
 
-	found, err := a.Find(ctx)
-	if err != nil {
-		return err
-	}
-	if !found {
-		return fmt.Errorf("GVK %s '%v' is not found", u.GroupVersionKind(), u.GetName())
+	for i := 0; i < maxRetries; i++ {
+		found, err := a.Find(ctx)
+		if err == nil && found {
+			op := directbase.NewDeleteOperation(c, u)
+			_, err = a.Delete(ctx, op)
+			if err == nil {
```

Comment:
Is it possible that err != nil but the deletion is still completed? I remember some resources may return an error during creation but still create the resource.

In that case we'll get err == nil && !found, and return an error in the second iteration, right? So maybe let only do err == nil && !found check for the first iteration?

Reply:
Oh interesting I didn't know that -- I can adjust the code to function like that!

Reply:
It may not be a common case. But I think NotFound in general can be considered fine when the resource was found in the first deletion.